source/common/http/status.cc

```diff
@@ -41,9 +42,13 @@ struct PrematureResponsePayload : public EnvoyStatusPayload {
 };
 
 template <typename T> void storePayload(absl::Status& status, const T& payload) {
-  absl::Cord cord(absl::string_view(reinterpret_cast<const char*>(&payload), sizeof(payload)));
+  const size_t payload_size = sizeof(payload);
+  char* buffer = new char[payload_size];
+  safeMemcpyUnsafeDst(buffer, &payload);
+  absl::Cord cord(absl::string_view(buffer, payload_size));
```

Comment:
The changes to this function do not solve any problem. I think the only change needed is something like this:

```
template <typename T> void storePayload(absl::Status& status, const T& payload) {
  const T* allocated = new T(payload);
  absl::Cord cord = absl::MakeCordFromExternal(absl::string_view(&allocated, sizeof(allocated)), [allocated]() { delete allocated; });
  status.SetPayload(EnvoyPayloadUrl, std::move(cord));
}
```

Then `getPayload` should not have to change at all since the object will already be correctly aligned.

Reply:
It is safe to cast to `const char *`.

Reply:
ok, trying that now in 68c52cfacc20ecff8f9e242c76d24ae616565256

Reply:
Looks like the tests pass with the latest changes, including with the address and undefined behavior sanitizers.